src/view/input_parsers/request_data.rs

```diff
@@ -62,6 +63,61 @@ mod parsers_request_items {
     use crate::app::services::request::entities::requests::BodyPayload;
     use crate::utils::regexes;
 
+    // Github Issue #8
+    // Parse "<key>:=[-]<number>"
+    // Parse "<key>:=true|false"
+    // Parse "<key>:='[<string>, <number>, <boolean>, <array>, <object>]'"
+    // Parse "<key>:='{"<key>": <string|number|boolean|array|object>"}'"
+    pub fn non_string_json_value(
+        s: &str,
+        base_request: &PartialRequestData,
+    ) -> Option<PartialRequestData> {
+        // regex match the number field and the boolean field
+        let mut re = regexes::request_items::single_json_value();
+        let mut matcher = re.captures(s)?;
+
+        let key = matcher.name("key")?.as_str();
+
+        // Parse the number or boolean
+        let mut value = match BodyPayload::from_str(matcher.name("value")?.as_str()) {
+            BodyPayload::Json(v) => {
+                if v.is_boolean() || v.is_number() {
+                    Some(v)
+                } else {
+                    None
+                }
+            }
+            BodyPayload::Raw(_) => None,
+        };
+
+        if value.is_none() {
+            // regex match the complex array and the object
+            re = regexes::request_items::combine_json_value();
+            matcher = re.captures(s)?;
+
+            // Parse the complex array or object
+            value = match BodyPayload::from_str(matcher.name("value")?.as_str()) {
+                BodyPayload::Json(v) => Some(v),
+                _ => return None,
+            }
+        }
```

Comment:
You could replace this by a more functional approach....
```rs
let value: &str = matcher.name("value")?.as_str();

let value: BodyPayload = serde_json::from_str<Value>(value).ok()
   .map(|v| BodyPayload::Json(v))
  .or_else(|| {
      // If the serde_json::from_str has failed. So it tries the other way, without the single quotes

       re = regexes::request_items::combine_json_value();
        matcher = re.captures(value)?; // This is the value of first line of this block of code

       let value: Value = serde_json::from_str<Value>(value).ok()?;
      Some(BodyPayload::Json(v))
  });
```

I guess this `or_else` could be splited in smaller code, but this make the code more readable.

Another thing is change the regex of second match for a regex to check only the "value". Instead check all input.

This regex would be something like just `'(?<value>[ -~]+)'` (this only check if a string start and end with `'`, and get the value in middle)

Reply:
When user type `married:="false"`, it will parse to `Value::String("false")`. If we check it, it can prevent all non boolean and number field.

Reply:
Any reason use `serve_json::from_str` rather than `BodyPayload::from_str`?

Reply:
@talis-fb 

If the `check_complex` is fine, I may change the `check_complex` to be a function.

`
        let check_complex = |value| {
            let re = regexes::request_items::combine_json_value();
            let matcher = re.captures(value)?;

            // Parse the complex array or object
            serde_json::from_str::<Value>(matcher.name("value")?.as_str())
            .ok()
        };
        // regex match the number field and the boolean field
        let re = regexes::request_items::non_string_body_value();
        let matcher = re.captures(s)?;

        let key = matcher.name("key")?.as_str();
        let value = matcher.name("value")?.as_str();

        let value = serde_json::from_str::<Value>(value).map_or_else(
            |_| check_complex(value),
            |v| {
                if v.is_boolean() || v.is_number() {
                    Some(v)
                } else {
                    check_complex(value)
                }
            },
        )?;

        let mut request = base_request.clone();
        
        request.body = match request.body {
            Some(BodyPayload::Json(serde_json::Value::Object(mut json))) => {
                json.insert(key.to_string(), value);
                BodyPayload::Json(serde_json::Value::Object(json))
            }
            _ => BodyPayload::Json(serde_json::json!({key: value})),
        }
        .into();

        Some(request)
`

Reply:
@SummerGram futhermore, It is better add some unit tests for this case I said like `key:={}`, `key:=[]`, `key:="{}"`, `key:="false"`, `key:={"name":"Talis"}`